internal/cmprefimpl/main.go

```diff
@@ -115,3 +116,16 @@ func deduplicateGeometries(geoms []geom.Geometry) []geom.Geometry {
 	})
 	return geoms
 }
+
+// 3D and Measure geometries are skipped because the C bindings for libgeos
+// doesn't fully support Z and M value.
+func remove3DAndMeasureGeometries(geoms []geom.Geometry) []geom.Geometry {
```

Comment:
Would it ever be needed to strip the Z and M coordinates rather than skip whole Z or M geometries?

Reply:
That's a good point. If we strip Z and M instead of discarding the entire geometry, we will end up with more test cases which is always a good thing. Fixed.